### PR TITLE
Followup fixes for Bug 1426464

### PR DESCRIPTION
**Pull request by rlr**
r? @Mardak

### CODE REVIEW

File: system-addon/content-src/components/TopSites/TopSitesEdit.jsx [hunk 1]
```diff
@@ -71,21 +71,12 @@ export class _TopSitesEdit extends React.PureComponent {
...
-        editLabel = this.props.TopSites.rows[editIndex].label;
-        editUrl = this.props.TopSites.rows[editIndex].url;
-      } else {
-        editIndex =  this.props.TopSites.editForm.site.index;
-        editLabel = "";
-        editUrl = "";
+      if (!editIndex || editIndex < 0) {
+        editIndex =  this.props.TopSites.editForm.index;
       }
+      editLabel = this.props.TopSites.rows[editIndex] ? this.props.TopSites.rows[editIndex].label || this.props.TopSites.rows[editIndex].hostname : "";
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSitesEdit.jsx]: Oh, I actually meant for both label and url. Maybe
```js
const editIndex = this.state.editIndex >= 0 ? this.state.editIndex : this.props.TopSites.editForm.index;
const editSite = this.props.TopSites.rows[editIndex] || {};
…
  label={editSite.label || editSite.hostname}
  url={editSite.url}
```

**Comment by rlr** [comment]
still have a failing test...

File: system-addon/content-src/components/TopSites/TopSitesEdit.jsx [hunk 2]
```diff
@@ -133,8 +121,8 @@ export class _TopSitesEdit extends React.PureComponent {
           <div className="modal-overlay" onClick={this.onModalOverlayClick} />
           <div className="modal">
             <TopSiteForm
-              label={editLabel}
-              url={editUrl}
+              label={editSite.label || editSite.hostname || ""}
+              url={editSite.url || ""}
```
Comment (rlr) [inline system-addon/content-src/components/TopSites/TopSitesEdit.jsx]: I had to add the '|| ""' because otherwise the component would keep it's old props.

**Comment by rlr** [comment]
@Mardak ok, this probably wants more tests but I think/hope it's good now.

**Comment by as-pine-proxy** [comment]
[Treeherder: e0536180cfcf073c9883ff870e9b16dfe13c369f](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=e0536180cfcf073c9883ff870e9b16dfe13c369f)

**Pull request merged**